pkg/runs/runsGet.go

```diff
@@ -316,39 +326,40 @@ func GetRunsFromRestApi(
 
 		for !gotAllResults && err == nil {
 
-            log.Printf("Requesting page '%d' ", pageNumberWanted)
+			log.Printf("Requesting page '%d' ", pageNumberWanted)
 
 			var runData *galasaapi.RunResults
-            runData, err = getRunsPageFromRestApi(
-                pageCursor,
-                runName,
-                requestorParameter,
-                resultParameter,
-                fromAgeMins,
-                toAgeMins,
-                fromTime,
-                toTime,
-                shouldGetActive,
-                apiClient,
-                restApiVersion,
-            )
-
-            if err == nil {
-                // Add all the runs into our set of results.
-                // Note: The ... syntax means 'all of the array', so they all get appended at once.
-                runsOnThisPage := runData.GetRuns()
-                results = append(results, runsOnThisPage...)
-
-                log.Printf("total runs: %v", len(results))
-
-                // Have we processed the last page ?
-                if !runData.HasNextCursor() || len(runsOnThisPage) < int(runData.GetPageSize()) {
-                    gotAllResults = true
-                } else {
-                    pageCursor = runData.GetNextCursor()
-                    pageNumberWanted++
-                }
-            }
+			runData, err = getRunsPageFromRestApi(
+				pageCursor,
+				runName,
+				requestorParameter,
+				resultParameter,
+				fromAgeMins,
+				toAgeMins,
+				fromTime,
```

Comment:
I do wonder if we're missing some higher-level object so we don't have to pass all these as separate parameters. More O-O would mean fewer parameters to pass. 
Maybe "RunsQuery" with a builder pattern ? runsQuery.getNextPage()...etc. 
At what point do we say this is too many I wonder ?

Reply:
Good point - addressed in https://github.com/galasa-dev/cli/pull/321/commits/116452c69e361c3cbae6cee4d76df6449503b56c